### PR TITLE
Implementation of new read-marker semantics

### PR DESCRIPTION
**Pull request by richvdh**
Separate the read-up-to marker from the read-receipts, and show a status
bar when the read marker is above the screen. Move the read-marker down
when it is on the screen and there is user activity.

(This requires corresponding changes in vector-web, to provide the CSS and
img)

This PR builds on https://github.com/matrix-org/matrix-react-sdk/pull/158 and https://github.com/matrix-org/matrix-react-sdk/pull/159. See b34fd86 for the relevant changes here.

### CODE REVIEW

File: src/components/structures/TimelinePanel.js [hunk 1]
```diff
@@ -401,6 +542,21 @@ module.exports = React.createClass({
         return this.props.room.getEventReadUpTo(myUserId);
     },
 
+    _setReadMarker: function(eventId, eventTs) {
+        // ideally we'd sync these via the server, but for now just stash them
```
Comment (ara4n) [inline src/components/structures/TimelinePanel.js]: might want to tag this as TODO in the comment

**Comment by ara4n** [comment]
this is a bit of an epic, but from a high level read through it all looks very plausible. lgtm

**Comment by richvdh** [comment]
@ara4n, did you want to take another look now that the prerequisite branches have landed, or shall I just merge it into rav/roomview_works?

**Comment by richvdh** [comment]
well, merging it then.

**Pull request merged**